modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/exp/RexImpTable.java

```diff
@@ -3077,6 +3077,16 @@ private static class BinaryImplementor extends AbstractRexCallImplementor {
         argValueList = FlatLists.append(argValueList, fieldComparator);
       }
 
+      if (type0 == BigDecimal.class && type1 == BigDecimal.class && op == DIVIDE) {
```

Comment:
doesn't look reliable. There is still exists path which handles division in an old way (see org.apache.ignite.internal.sql.engine.exec.exp.IgniteExpressions#divideExact)

Reply:
`AbstractRexCallImplementor::harmonize` ensures that both operands have identical type. 
e.g. in a / b where a is int and b is decimal, a is going to be converted to decimal.

Reply:
yeah, but

```
      defineMethod(IgniteSqlOperatorTable.DECIMAL_DIVIDE, IgniteMethod.DECIMAL_DIVIDE.method(), NullPolicy.ARG0);
      
      ...

    private void defineMethod(SqlOperator operator, Method method,
        NullPolicy nullPolicy) {
      map.put(operator, new MethodImplementor(method, nullPolicy, false));
    }

    ...

     MethodImplementor(Method method, NullPolicy nullPolicy,
        boolean harmonize) {
```

we don't even have test to make sure harmonization is working as expected

Reply:
@korlov42 I definitely works. 

I have updated `genValueStatement` in `RexImpTable` with the following code

```
  private static String exprToString(List<Expression> exprs) {
      return exprs.stream()
              .map(e -> e.getType() + " >>" + e + "<<")
              .collect(Collectors.joining());
    }
    
   private ParameterExpression genValueStatement(
      ..
    if (harmonize) {
        System.err.println("HARMONIZE " + call + " " + exprToString(optimizedArgValueList));
        optimizedArgValueList =
            harmonize(optimizedArgValueList, translator, call);
        System.err.println("HARMONIZE " + call + " RESULT " + exprToString(optimizedArgValueList));
      }
```

When I run the following code:
```
@Test
    public void test() {
        sql("CREATE TABLE t(id INT PRIMARY KEY, int_col INT, decimal_col DECIMAL(3,1) )");

        sql("INSERT INTO t VALUES (1, 2, 2.1)");

        sql("SELECT int_col / decimal_col FROM t");
    }
```
I see 

```
HARMONIZE /($t0, $t1) class java.lang.Integer >>input_value<<class java.math.BigDecimal >>input_value0<<
HARMONIZE /($t0, $t1) RESULT class java.math.BigDecimal >>input_value == null ? (java.math.BigDecimal) null : new java.math.BigDecimal(
  input_value.intValue())<<class java.math.BigDecimal >>input_value0<<
```
int, big decimal becomes big decimal, big decimal.

Reply:
I see where I was wrong: we use `MethodImplementor` for `DECIMAL_DEVIDE`, but changes in question in `BinaryImplementor` and `harmonize` flag is indeed set to `true`